### ISSUE TITLE
Cannot add to cart after passing an order if cache is enabled

### ISSUE THREAD

**Issue by Tofandel** [PrestaShop/PrestaShop#36474]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

With memcached enabled, when a customer passes an order and then goes back to the shop, adding any item to the cart fails, the add to cart request returns a `Redirect 302` to the home page of the site instead of adding to the cart

If memached is disabled then this issue does not happen

### Steps to reproduce

1. Go in the admin to Advanced > Performances
2. Enable CacheMemcached
3. Go to the frontend and pass an order with a logged in account
4. Once the order is done, go to the shop and try to add another product to the cart

### PrestaShop version(s) where the bug happened

8.1.6

### PHP version(s) where the bug happened

8.2

**Comment by paulnoelcholot** [comment]
Hello,

Thanks for reporting this issue!
Ping @PrestaShop/qa-functional  : Could someone please try to reproduce the issue?

Thanks in advance!

**Comment by kpodemski** [comment]
Hello @Tofandel 

If you've used PHP 8.2 with PrestaShop 8.1 it could produce some side-effects since PrestaShop 8.1 **does not** support PHP 8.2.

I am also unsure about the exact steps to follow:
> Once the order is done, go to the shop and try to add another product to the cart

You mean, on the account who previously ordered something, you want to start a new order?

**Comment by Tofandel** [comment]
> You mean, on the account who previously ordered something, you want to start a new order?

Yes, with still the same session, if after an order you try to add something to the cart, the cart requests sends a 302 instead of the usual add to cart success. It's the same with php 8.1 or 8.2

Once you clear the session cookies you can add to the cart again (or if you disable the memcached module)

**Comment by kpodemski** [comment]
I'm sure you'll encounter more issues using Memcache, so for now, I recommend using some third-party modules to help you optimize your performance. There are some modules which implement Redis better, there are some which implements full page cache. Memcache hasn't been the priority in the project for a long time, and I doubt it will be in the near future.

**Comment by Tofandel** [comment]
Okay it seems Memcache is not the cause in the end, it's still disabled and this issue is still happening until I clear the prestashop cache

**Comment by florine2623** [comment]
Hello @Tofandel ,

I tried without Memcache enabled and I can't reproduce your issue on my end.

https://github.com/PrestaShop/PrestaShop/assets/16019289/11635984-7410-433d-a3a3-f71322bfe6ff

I will close your issue as I cannot reproduce it.
Thanks!

**Comment by Tofandel** [comment]
Could you try with the locale set to french, so the cart url is not the usual /cart but /panier, maybe it has something to do with it

**Comment by kpodemski** [comment]
@Tofandel you can try it yourself, you can also try to disable Memcache, adding to cart is something that is tested by numerous automated tests

**Comment by Tofandel** [comment]
I disabled memcache already, the issue is already happening if the PS cache is enabled

I'm sure adding to cart is tested, but the issue is only after ordering and trying to add to cart again which I'm sure is not covered

This means none of our clients can pass a new order right now even years after passing one (because they can't add anything to their cart)

I tried to look into where it's happening but the code is very hard to follow as it's not just concentrated in one place, there is hooks and a lot of logic in different controllers

I will try with the shop in EN to see if it has anything to do with that

**Comment by Tofandel** [comment]
Yes issue is not happening in EN, only FR

**Comment by kpodemski** [comment]
I don't have such issue with French language and the latest PrestaShop version.

**Comment by Tofandel** [comment]
Since I enabled the EN pack, now the issue is not happening in FR anymore. This is driving me mad, but I guess now the issue is gone for good

I really have no idea what could cause this
